paddle/framework/backward.md

```diff
@@ -2,9 +2,20 @@
 
 ## Motivation
 
-In Neural Network, the backpropagation algorithm follows the chain rule, so we need to compound the gradient operators/expressions together with the chain rule. Every forward network needs a backward network to construct the full computation graph, the operator/expression's backward pass will be generated respect to forward pass.
+In Neural Network, many model is solved by the the backpropagation algorithm(known as BP) at present. Technically it caculates the gradient of the loss function, then distributed back through the networks. Follows the chain rule, so we need a module chains the gradient operators/expressions together with to construct the backward pass. Every forward network needs a backward network to construct the full computation graph, the operator/expression's backward pass will be generated respect to forward pass. 
 
-## Backward Operator Registry
+## Implementation
+
+In this design doc, we exported only one API for generating the backward pass.
+
+```c++
+std::unique_ptr<OperatorBase> Backward(const OperatorBase& forwardOp,
+    const std::unordered_set<std::string>& no_grad_vars);
+```
+
+The implementation behind it can be divided into two parts, ** Backward Operator Creating** and **Backward Operator Building**.
+
+###Backward Operator Registry
 
 A backward network is built up with several backward operators. Backward operators take forward operators' inputs outputs, and output gradients and then calculate its input gradients.
```

Comment:
take forward operators' inputs, outputs, "output gradients" it's inputs.

Reply:
Does "forward operator's output gradients" means the "input grad" of next forward operator connected to it? Or the cost function's output?

Reply:
fix Done.